mne/encoding/feature.py

```diff
@@ -0,0 +1,250 @@
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    time_window : array, shape (2,)
+        Specifies a minimum/maximum delay in seconds. In this case, all
+        delays between `min(time_window)` and `max(time_window)`
+        will be calculated using `sfreq`.
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.
+    """
+    def __init__(self, delays=None, time_window=None, sfreq=1.):
+        # Check if we need to creafte delays ourselves
+        if time_window is not None:
+            if sfreq is None:
+                raise ValueError('If time_window given, must give sfreq.')
+            if delays is not None:
+                raise ValueError('Delays must be None if time_window is given')
+            tmin, tmax = time_window
+            start_idx = int(round(tmin * sfreq))
+            delays = np.arange(start_idx, int(round(tmax * sfreq)) + 1) / sfreq
+        elif delays is None:
+            raise ValueError('One of `time_window` or `delays` must be given.')
+        self.delays = np.asarray(delays)
+        self.sfreq = float(sfreq)
+        self.feature_type = 'delay'
+
+    def fit(self, X, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+        """
+        X_del, names = _delay_timeseries(X, self.delays, sfreq=self.sfreq)
+        self.X_delayed_ = X_del
+        self.names_ = names
+
+    def transform(self, X=None, y=None):
+        """See `fit` method."""
+        return self.X_delayed_
+
+    def fit_transform(self, X, y=None):
+        """See `fit` method."""
+        self.fit(X)
+        return self.transform()
+
+
+class EventsBinarizer(object):
+    """Create a continuous-representation of event onsets.
+
+    Parameters
+    ----------
+    n_times : int
+        The total number of times in the output array.
+    sfreq : float
+        The sampling frequency used to convert times to sample indices.
+        Defaults to 1.
+    sparse : bool
+        Whether to output continuous events as a dense array or a sparse
+        matrix. Defaults to False.
+    """
+    def __init__(self, n_times, sfreq=1., sparse=False):
+        self.n_times = n_times
+        self.sfreq = sfreq
+        self.feature_type = 'event_id'
```

Comment:
if event_id all the time, no need to save it as a param (plus this will crash if you clone the object)